smokefree/sites/all/modules/custom/icfi_api_keys/ICFIApi.inc

```diff
@@ -0,0 +1,104 @@
+<?php
+
+
+class ICFIApi {
+  protected $machineName;
+  protected $apiValue;
+  protected $isJsVariable;
+
+  public function __construct($machineName, $apiValue, $isJsVariable) {
+    if (empty($machineName) || empty($apiValue)) {
+      throw new InvalidArgumentException('Please supply a valid api value and an appropriate machine name');
+    }
+
+    // Remove accidental whitespaces at the end
+    $this->machineName = trim($machineName);
+    $this->apiValue = trim($apiValue);
+    $this->isJsVariable = $isJsVariable;
+  }
+
+  /**
+   * Create a new API key entry in the database
+   */
+  public function create() {
+
+    // Check if this machine name exists, only enter a new record if there is no existing value
+    $entryExists = self::getKeyValue($this->machineName);
+    if (empty($entryExists)) {
+      $transaction = db_transaction();
+      try {
+        $query = db_insert('icfi_api_keys')->fields(array(
+            'api_key_machine_name' => $this->machineName,
+            'api_key_value' => $this->apiValue,
+            'javascript_variable' => $this->isJsVariable,)
+        );
+        $query->execute();
+        return TRUE;
+      }
+      //Shouldn't really happen
+      catch (Exception $e) {
+        $transaction->rollback();
+        watchdog_exception('icfi_api_keys', $e);
+        return FALSE;
+      }
+    }
+    else {
+      return FALSE;
+    }
+  }
+
+  /**
+   * Updates an existing API key entry in the database
+   */
+  public function update() {
+    $transaction = db_transaction();
```

Comment:
What happens if the key already exists in the table? Should you check for that here?

Reply:
This function should be called only if the key exists. However, I realized that I should probably check if the value exists before attempting to update. I should probably return a "FALSE" if the entry does not exist